mypy/test/test_check.py

```diff
@@ -122,13 +122,12 @@ def run_test_once(self, testcase: DataDrivenTestCase, incremental=0) -> None:
             a = e.messages
         a = normalize_error_messages(a)
 
-        if output != a and mypy.myunit.UPDATE_TESTCASES:
+        if output != a and pytest.config.getoption('UPDATE_TESTCASES'):
             update_testcase_output(testcase, a, mypy.myunit.APPEND_TESTCASES)
 
-        assert_string_arrays_equal(
-            output, a,
+        assert output == a, \
             'Invalid type checker output ({}, line {})'.format(
-                testcase.file, testcase.line))
+                testcase.file, testcase.line)
```

Comment:
Yeah, a unified context diff (like `diff -u` and the `git diff` default) would be great.

I've actually never tried the `-i` and `-u` options. They don't seem to particularly be documented, either, though I can infer what I think they're supposed to do from this:
```
        elif a == '-u':
            APPEND_TESTCASES = '.new'
            UPDATE_TESTCASES = True
        elif a == '-i':
            APPEND_TESTCASES = ''
            UPDATE_TESTCASES = True
```
(and from a vague recollection of hearing about that kind of functionality in myunit.)